momento-sdk/src/main/java/momento/sdk/CacheClient.java

```diff
@@ -677,6 +681,219 @@ public CompletableFuture<CacheSetFetchResponse> setFetch(String cacheName, Strin
+   * @param elements - The value to score map to add to the sorted set.
+   * @return Future containing the result of the put elements operation.
+   */
+  public CompletableFuture<CacheSortedSetPutElementsResponse> sortedSetPutElements(
+      String cacheName, String sortedSetName, Map<String, Double> elements) {
+    return scsDataClient.sortedSetPutElements(cacheName, sortedSetName, elements, null);
+  }
+
+  /**
+   * Adds elements to the given sorted set. If the elements already exist, their scores are updated.
+   * Creates the sorted set if it does not exist.
+   *
+   * @param cacheName - The cache containing the sorted set.
+   * @param sortedSetName - The sorted set to add to.
+   * @param elements - The value to score map to add to the sorted set.
+   * @param ttl TTL for the set in cache. This TTL takes precedence over the TTL used when
+   *     initializing a cache client. Defaults to client TTL.
+   * @return Future containing the result of the put elements operation.
+   */
+  public CompletableFuture<CacheSortedSetPutElementsResponse> sortedSetPutElementsByteArray(
+      String cacheName,
+      String sortedSetName,
+      Map<byte[], Double> elements,
+      @Nullable CollectionTtl ttl) {
+    return scsDataClient.sortedSetPutElementsByteArray(cacheName, sortedSetName, elements, ttl);
+  }
+
+  /**
+   * Adds elements to the given sorted set. If the elements already exist, their scores are updated.
+   * * Creates the sorted set if it does not exist.
+   *
+   * @param cacheName - The cache containing the sorted set.
+   * @param sortedSetName - The sorted set to add to.
+   * @param elements - The value to score map to add to the sorted set.
+   * @return Future containing the result of the put elements operation.
+   */
+  public CompletableFuture<CacheSortedSetPutElementsResponse> sortedSetPutElementsByteArray(
+      String cacheName, String sortedSetName, Map<byte[], Double> elements) {
+    return scsDataClient.sortedSetPutElementsByteArray(cacheName, sortedSetName, elements, null);
+  }
+
+  /**
+   * Fetch the elements in the given sorted set by index (rank).
+   *
+   * @param cacheName - The cache containing the sorted set.
+   * @param sortedSetName - The sorted set to fetch from.
+   * @param startRank - The rank of the first element to fetch. Defaults to 0. This rank is
+   *     inclusive, i.e. the element at this rank will be fetched.
+   * @param endRank - The rank of the last element to fetch. This rank is exclusive, ie the element
+   *     at this rank will not be fetched. Defaults to null, which fetches up until and including
+   *     the last element.
+   * @param order - The order to fetch the elements in. Defaults to ascending.
+   * @return Future containing the result of the fetch operation.
+   */
+  public CompletableFuture<CacheSortedSetFetchResponse> sortedSetFetchByRank(
+      String cacheName,
+      String sortedSetName,
+      @Nullable Integer startRank,
+      @Nullable Integer endRank,
```

Comment:
Optionals in Java as method parameters generally feel off to me. They don't add null safety and they require an extra layer of wrapping to use. I know they were explicitly not idiomatic when optional was released. My instincts might be wrong though.

Reply:
okay.  I don't consider myself up-to-date on java best practices on this, and a quick google search yielded articles like [this one](https://medium.com/javarevisited/4-reasons-why-you-should-use-java-optional-or-not-4e44d51a9645) which agree with your take.  So I am happy to go with your instinct here.  It might be a thing to call out explicitly in your bug bash doc though to see if others with Java backgrounds want to weigh in on it.